plugin/pkg/scheduler/core/equivalence_cache.go

```diff
@@ -61,74 +66,153 @@ func NewEquivalenceCache(getEquivalencePodFunc algorithm.GetEquivalencePodFunc)
 	}
 }
 
-// addPodPredicate adds pod predicate for equivalence class
-func (ec *EquivalenceCache) addPodPredicate(podKey uint64, nodeName string, fit bool, failReasons []algorithm.PredicateFailureReason) {
+func (ec *EquivalenceCache) UpdateCachedPredicateItem(pod *v1.Pod, nodeName, predicateKey string, fit bool, reasons []algorithm.PredicateFailureReason, equivalenceHash uint64) {
```

Comment:
We dont need unit tests for each function (ideal but not necessary) but can you have a ExampleTest function like https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/cmd_test.go#L148